add hosting trial product to plans list

vykes-mac opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Closes https://github.com/Automattic/dotcom-forge/issues/4043

## Proposed Changes

* Add hosting trial plan when `plans/hosting-trial `is enabled
* Add hosting trial plan to available list for:
1.  `PLAN_BUSINESS`
2. `PLAN_BUSINESS_2_YEARS`
3. `PLAN_BUSINESS_3_YEARS`
4. `PLAN_ECOMMERCE`
5. `PLAN_ECOMMERCE_MONTHLY`
6. `PLAN_ECOMMERCE_2_YEARS`
7. `PLAN_ECOMMERCE_3_YEARS`

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

*  Force plan at  `checkout/siteSlug/wp_bundle_hosting_trial_monthly`. You should see the plan name in the cart.
![image](https://github.com/Automattic/wp-calypso/assets/47489215/29789dbe-d205-43e6-9a74-82d21ee4781a)


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

vykes-mac committed:
add hosting trial product to plans list

vykes-mac committed:
add description and tagline

vykes-mac committed:
fix failing test

p-jackson commented on client/my-sites/plans/current-plan/index.jsx:
```diff
@@ -220,7 +221,8 @@ class CurrentPlan extends Component {
 		const currentPlanSlug = selectedSite?.plan?.product_slug ?? '';
 		const isEcommerceTrial = currentPlanSlug === PLAN_ECOMMERCE_TRIAL_MONTHLY;
 		const isBusinessTrial = currentPlanSlug === PLAN_MIGRATION_TRIAL_MONTHLY;
```
It doesn't look like `isBusinessTrial` constant doesn't look like it's used anywhere else. But I think both the hosting and migration trial should count as "business trial". That was sort of the original conception.

```suggestion
		const isBusinessTrial = currentPlanSlug === PLAN_MIGRATION_TRIAL_MONTHLY || currentPlanSlug === PLAN_HOSTING_TRIAL_MONTHLY`
```

p-jackson commented on packages/calypso-products/src/plans-list.tsx:
```diff
@@ -3609,3 +3619,20 @@ if ( isEnabled( 'plans/migration-trial' ) ) {
+		term: TERM_MONTHLY,
+		getBillingTimeFrame: () => i18n.translate( 'free trial' ),
+		getStoreSlug: () => PLAN_HOSTING_TRIAL_MONTHLY,
+		getTitle: () => i18n.translate( 'Hosting Trial' ),
```
There's some discussion about the plan title here pet6gk-Ik-p2#comment-596, but not blocking. We can change this later.

p-jackson commented on packages/calypso-products/src/plans-list.tsx:
```diff
@@ -3609,3 +3619,20 @@ if ( isEnabled( 'plans/migration-trial' ) ) {
 	};
 }
+
+if ( isEnabled( 'plans/hosting-trial' ) ) {
```
I don't feel like we need the feature flag here. It seems ok for Calypso to have support for the new constant even in production. All we want to guard against is the user seeing the UI that allows them to select the plan for now.

p-jackson reviewed (approved):
Some notes but this is g2g

p-jackson committed:
The hosting trial _is_ a business trial

p-jackson committed:
Don't exclude plan definition from Calypso code based on feature flag

There's no harm in having the plan defined. It just means that
everything will work for a12s who happen to be looking at their trial
sites on production.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9188941

Thank you @vykes-mac for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9188941).
